pipeline/customer/order_test.py

```diff
@@ -146,6 +146,121 @@ def test_invalid_total_price_type(self):
+        data = []
+        expected = []
+
+        with TestPipeline() as p:
+            actual = (
+                p
+                | beam.Create(data)
+                | GroupByCustomer()
+            )
+            assert_that(actual, equal_to(expected))
+
+    def test_single(self):  # pylint: disable=no-self-use
+        data = [(880, {
+            'id': 880,
+            'customer_id': 1,
+            'total_price': '287.69',
+        })]
+        expected = [(1, [{
+            'id': 880,
+            'customer_id': 1,
+            'total_price': '287.69',
+        }])]
+
+        with TestPipeline() as p:
+            actual = (
+                p
+                | beam.Create(data)
+                | GroupByCustomer()
+            )
+            assert_that(actual, equal_to(expected))
+
+    def test_multiple(self):  # pylint: disable=no-self-use
+        data = [
+            (3607, {
+                'id': 3607,
+                'customer_id': 3,
+                'total_price': '373.02',
+            }),
+            (3949, {
+                'id': 3949,
+                'customer_id': 3,
+                'total_price': '702.88',
+            }),
+        ]
+        expected = [(3, [
+            {
+                'id': 3607,
+                'customer_id': 3,
+                'total_price': '373.02',
+            },
+            {
+                'id': 3949,
+                'customer_id': 3,
+                'total_price': '702.88',
+            },
+        ])]
+
+        with TestPipeline() as p:
+            actual = (
+                p
+                | beam.Create(data)
+                | GroupByCustomer()
+            )
+            assert_that(actual, equal_to(expected))
+
+    def test_mixed(self):  # pylint: disable=no-self-use
+        data = [
+            (880, {
+                'id': 880,
+                'customer_id': 1,
+                'total_price': '287.69',
+            }),
+            (3607, {
+                'id': 3607,
+                'customer_id': 3,
+                'total_price': '373.02',
+            }),
+            (3949, {
+                'id': 3949,
+                'customer_id': 3,
+                'total_price': '702.88',
+            }),
+        ]
+        expected = [
+            (1, [{
+                'id': 880,
+                'customer_id': 1,
+                'total_price': '287.69',
+            }]),
+            (3, [
+                {
+                    'id': 3607,
+                    'customer_id': 3,
+                    'total_price': '373.02',
+                },
+                {
+                    'id': 3949,
+                    'customer_id': 3,
+                    'total_price': '702.88',
+                },
+            ]),
+        ]
+
+        with TestPipeline() as p:
+            actual = (
+                p
+                | beam.Create(data)
+                | GroupByCustomer()
+            )
+            assert_that(actual, equal_to(expected))
```

Comment:
Maybe it makes sense to put 255:261 into a separated helper function. On the other hand I like this "bare" stuff here.